### PR TITLE
Added support for comma separated appproject label

### PR DESCRIPTION
**Pull request by SamMHD**
Changed cache structure to support for Multi-Link and reverse fetch over AppProjects and Namespaces in order to support multi appproject for a single namespace.

After these changes we can set a comma-separated list in label for each namespace

### CODE REVIEW

File: controllers/namespace_controller.go [hunk 1]
```diff
@@ -40,34 +41,90 @@ import (
...
-	m  map[string]string
+	mu          sync.Mutex
+	projects    map[string]AppProjectNameset
+	namespaces  map[string]NamespaceNameset
+	initialized bool
 }
 
-// Inc increments the counter for the given key.
-func (c *SafeNsCache) Set(k, v string) {
+// LeaveProject will add given namespace from given project in SafeNsCache entries
```
Comment (ssttehrani) [inline controllers/namespace_controller.go:65]: Could you please update the comment?

**Comment by ssttehrani** [comment]
LGTM, Thank you dear @SamMHD

**Review by ssttehrani — APPROVED** [review]

File: controllers/namespace_controller.go (hunk 1 above)
Comment (SamMHD) [reply controllers/namespace_controller.go:65]: What should I change about it?

Comment (SamMHD) [reply controllers/namespace_controller.go]: Done, Thank you for mentioning it.

**Comment by SamMHD** [comment]
@ssttehrani, thank you for review. How can we push it forward?

**Review by pedy4000 — APPROVED** [review]

**Pull request merged**